framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/ServiceProxying/CSharp/CSharpServiceProxyGenerator.cs

```diff
@@ -57,167 +91,198 @@ public class CSharpServiceProxyGenerator : ServiceProxyGeneratorBase<CSharpServi
         "using Volo.Abp.Http.Client.ClientProxying;"
     };
 
+    private static readonly List<string> InterfaceUsingNamespaceList = new()
+    {
+        "using System;",
+        "using System.Threading.Tasks;",
+        "using Volo.Abp.Application.Dtos;",
+        "using Volo.Abp.Application.Services;"
+    };
+
     public CSharpServiceProxyGenerator(
         CliHttpClientFactory cliHttpClientFactory,
         IJsonSerializer jsonSerializer) :
         base(cliHttpClientFactory, jsonSerializer)
     {
     }
 
-    public async override Task GenerateProxyAsync(GenerateProxyArgs args)
+    public override async Task GenerateProxyAsync(GenerateProxyArgs args)
     {
         CheckWorkDirectory(args.WorkDirectory);
         CheckFolder(args.Folder);
 
         if (args.CommandName == RemoveProxyCommand.Name)
         {
-            RemoveClientProxyFile(args);
+            var folder = args.Folder.IsNullOrWhiteSpace() ? ProxyDirectory : args.Folder;
+            var folderPath = Path.Combine(args.WorkDirectory, folder);
+
+            if (Directory.Exists(folderPath))
+            {
+                Directory.Delete(folderPath, true);
+            }
+
+            Logger.LogInformation($"Delete {GetLoggerOutputPath(folderPath, args.WorkDirectory)}");
             return;
         }
 
-        var applicationApiDescriptionModel = await GetApplicationApiDescriptionModelAsync(args);
+        var applicationApiDescriptionModel = await GetApplicationApiDescriptionModelAsync(args, new ApplicationApiDescriptionModelRequestDto
+        {
+            IncludeTypes = !args.WithoutContracts
+        });
+
+        foreach (var controller in applicationApiDescriptionModel.Modules.Values.SelectMany(x => x.Controllers)
+                     .Where(x => x.Value.Interfaces.Any() && ServicePostfixes.Any(s => x.Value.Interfaces.Last().Type.EndsWith(s))))
+        {
+            await GenerateClassFileAsync(args, controller.Value);
+        }
+
+        foreach (var controller in applicationApiDescriptionModel.Modules.Values.SelectMany(x => x.Controllers).Where(x => ShouldGenerateProxy(x.Value)))
```

Comment:
Looks like this foreach block is duplicate with above foreach block. They are doing the same thing